### ISSUE TITLE
Test error/segfault on macOS

### ISSUE THREAD

**Issue by andreasnoack** [timholy/Revise.jl#134]
On
```julia
Julia Version 1.0.0-DEV.65
Commit 2990f0e1f4 (2018-08-07 06:58 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin17.7.0)
  CPU: Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, haswell)
```
I'm currently getting this when running the tests of Revise master
```julia
[ Info: tracking Base
File paths: Error During Test at /Users/andreasnoack/.julia/dev/Revise/test/runtests.jl:366
  Test threw exception KeyError("/private/var/folders/3x/jb0wdqls3yq0lbnydx6tzkz00000gn/T/6qYwvW7VZA/PC/src/PC.jl")
  Expression: Revise.get_def(m) == convert(Revise.RelocatableExpr, $(Expr(:quote, :(($(Expr(:$, :fn1)))() = begin
          1
      end))))
  KeyError: key "/private/var/folders/3x/jb0wdqls3yq0lbnydx6tzkz00000gn/T/6qYwvW7VZA/PC/src/PC.jl" not found
  Stacktrace:
   [1] getindex at ./dict.jl:478 [inlined]
   [2] macro expansion at ./logging.jl:308 [inlined]
   [3] get_def(::Method) at /Users/andreasnoack/.julia/dev/Revise/src/Revise.jl:542
   [4] macro expansion at /Users/andreasnoack/julia-dev/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1067 [inlined]
   [5] macro expansion at /Users/andreasnoack/julia-dev/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1070 [inlined]
   [6] macro expansion at /Users/andreasnoack/.julia/dev/Revise/test/runtests.jl:70 [inlined]
   [7] macro expansion at /Users/andreasnoack/julia-dev/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1079 [inlined]
   [8] top-level scope at /Users/andreasnoack/.julia/dev/Revise/test/runtests.jl:53
[ Info: tracking Base
File paths: Error During Test at /Users/andreasnoack/.julia/dev/Revise/test/runtests.jl:366
  Test threw exception KeyError("/private/var/folders/3x/jb0wdqls3yq0lbnydx6tzkz00000gn/T/6qYwvW7VZA/NPC/src/NPC.jl")
  Expression: Revise.get_def(m) == convert(Revise.RelocatableExpr, $(Expr(:quote, :(($(Expr(:$, :fn1)))() = begin
          1
      end))))
  KeyError: key "/private/var/folders/3x/jb0wdqls3yq0lbnydx6tzkz00000gn/T/6qYwvW7VZA/NPC/src/NPC.jl" not found
  Stacktrace:
   [1] getindex at ./dict.jl:478 [inlined]
   [2] macro expansion at ./logging.jl:308 [inlined]
   [3] get_def(::Method) at /Users/andreasnoack/.julia/dev/Revise/src/Revise.jl:542
   [4] macro expansion at /Users/andreasnoack/julia-dev/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1067 [inlined]
   [5] macro expansion at /Users/andreasnoack/julia-dev/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1070 [inlined]
   [6] macro expansion at /Users/andreasnoack/.julia/dev/Revise/test/runtests.jl:70 [inlined]
   [7] macro expansion at /Users/andreasnoack/julia-dev/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1079 [inlined]
   [8] top-level scope at /Users/andreasnoack/.julia/dev/Revise/test/runtests.jl:53
Test Summary:                   | Pass  Error  Total
Revise                          |  277      2    279
  LineSkipping                  |    7             7
  Parse errors                  |    4             4
  Methods and signatures        |   59            59
  Comparison and line numbering |   34            34
  Display                       |    4             4
  File paths                    |   81      2     83
  @__FILE__                     |    2             2
  Module docstring              |    6             6
  Line numbers                  |    5             5
  New submodules                |    3             3
  Method deletion               |   40            40
  Pkg exclusion                 |    2             2
  Manual track                  |    4             4
  Auto-track user scripts       |    4             4
  Distributed                   |   12            12
  Git                           |    4             4
  Recipes                       |    6             6
  Cleanup                       |              No tests
ERROR: LoadError: Some tests did not pass: 277 passed, 0 failed, 2 errored, 0 broken.
in expression starting at /Users/andreasnoack/.julia/dev/Revise/test/runtests.jl:51

signal (11): Segmentation fault: 11
in expression starting at no file:0
git_strmap_lookup_index at /Users/andreasnoack/julia-dev/usr/lib/libgit2.dylib (unknown line)
git_mwindow_put_pack at /Users/andreasnoack/julia-dev/usr/lib/libgit2.dylib (unknown line)
pack_backend__free at /Users/andreasnoack/julia-dev/usr/lib/libgit2.dylib (unknown line)
git_odb_free at /Users/andreasnoack/julia-dev/usr/lib/libgit2.dylib (unknown line)
git_repository__cleanup at /Users/andreasnoack/julia-dev/usr/lib/libgit2.dylib (unknown line)
git_repository_free at /Users/andreasnoack/julia-dev/usr/lib/libgit2.dylib (unknown line)
close at /Users/andreasnoack/julia-dev/usr/share/julia/stdlib/v1.0/LibGit2/src/types.jl:1033
jfptr_close_7177 at /Users/andreasnoack/julia-dev/usr/lib/julia/sys.dylib (unknown line)
jl_apply at /Users/andreasnoack/julia-dev/src/./julia.h:1536 [inlined]
run_finalizer at /Users/andreasnoack/julia-dev/src/gc.c:115
jl_gc_run_finalizers_in_list at /Users/andreasnoack/julia-dev/src/gc.c:210
run_finalizers at /Users/andreasnoack/julia-dev/src/gc.c:245 [inlined]
jl_gc_run_all_finalizers at /Users/andreasnoack/julia-dev/src/gc.c:282
jl_atexit_hook at /Users/andreasnoack/julia-dev/src/init.c:260
jl_exit at /Users/andreasnoack/julia-dev/src/jl_uv.c:604
exit at ./initdefs.jl:28 [inlined]
_start at ./client.jl:425
true_main at /Users/andreasnoack/julia-dev/usr/bin/julia (unknown line)
main at /Users/andreasnoack/julia-dev/usr/bin/julia (unknown line)
Allocations: 30939566 (Pool: 30935409; Big: 4157); GC: 70
ERROR: Package Revise errored during testing
```
Don't know if this is related to #133

**Comment by timholy** [comment]
Thank you so much for checking! This answers my main question, revealing that the failures are not just a Travis artifact :frowning_face: . The failures don't actually affect Revise but they could be serious for  Rebugger. Could you try #133 and see if it still errors? (And if it doesn't, whether that depends on the debugging commit?)

The segfault is actually different: https://github.com/JuliaLang/julia/issues/28306. I think the fix for that is in LibGit2.

**Comment by andreasnoack** [comment]
The tests pass with #133

**Comment by timholy** [comment]
Did you by any chance try it stripping out the debug commit? If you're willing we can also sit down and work on this together.

**Comment by andreasnoack** [comment]
Just tried the individual commits and https://github.com/timholy/Revise.jl/pull/133/commits/b28b7723cac6b6426e78e3e588976c9c967a2a8b is the one that fixes the issue.

**Comment by timholy** [comment]
Closed by #133.
